packages/pangea-multipass/pangea_multipass/sources/gdrive/gdrive.py

```diff
@@ -410,3 +420,23 @@ def list_all_file_ids(creds: Credentials) -> List[str]:
                 break
 
         return file_ids
+
+    @staticmethod
+    def check_user_access(creds: Credentials, file_id: str, user_email: str) -> Optional[str]:
+        """
+        Check if a specific user has access to a Google Drive file.
+        
+        :return: Access level (e.g., "owner", "writer", "reader") or None if no access.
+        """
+
+        service = build("drive", "v3", credentials=creds)
+        try:
+            # List the file's permissions
+            permissions = service.permissions().list(fileId=file_id, fields="permissions").execute()
+            for permission in permissions.get("permissions", []):
+                if permission.get("emailAddress") == user_email:
+                    return permission.get("role")  # e.g., "owner", "writer", "reader"
+            return None
+        except Exception as e:
+            print(f"Error: {e}")
```

Comment:
Should we remove this since it looks like this error is gracefully handled? Or maybe downgrade it to just a warning? We should also probably using the `logging` module so that users can choose whether or not to see our logs at all.

Reply:
Yes, I'll remove it.